### ISSUE TITLE
lottie: [RuntimeError] memory access out of bounds

### ISSUE THREAD

**Issue by theashraf** [thorvg/thorvg#2664]
we're experiencing an issue in dotlottie player rendering this asset and would appreciate your help in determining whether the problem lies within the asset itself or if it's related to ThorVG. 

The error occurs when using the ThorVG viewer. I suspect it might be related to the image assets, but I'm not certain about the exact cause.

Please see the screenshot below for more details:

![Error Screenshot](https://github.com/user-attachments/assets/d314d53e-fe87-4879-acea-89dfedaeb121)

You can also review the lottie here: [hero_animation.json](https://github.com/user-attachments/files/16705357/hero_animation.json)

related issue -> https://github.com/LottieFiles/dotlottie-web/issues/328

**Comment by hermet** [comment]
expression corruption:

```
ecma_lcache_invalidate (object_p=object_p@entry=0x7ffff7f39e90 <jerry_global_heap+5744>, name_cp=name_cp@entry=1, prop_p=0x7ffff7f3a108 <jerry_global_heap+6376> "\267\267") at ../src/loaders/lottie/jerryscript/jerry-core/ecma/base/ecma-lcache.cpp:205
205         if (entry_p->id != 0 && entry_p->prop_p == prop_p)
(gdb) where
#0  ecma_lcache_invalidate (object_p=object_p@entry=0x7ffff7f39e90 <jerry_global_heap+5744>,
    name_cp=name_cp@entry=1, prop_p=0x7ffff7f3a108 <jerry_global_heap+6376> "\267\267")
    at ../src/loaders/lottie/jerryscript/jerry-core/ecma/base/ecma-lcache.cpp:205
#1  0x00007ffff7e4ad46 in ecma_gc_free_property (options=0, prop_pair_p=0x7ffff7f3a108 <jerry_global_heap+6376>,
    object_p=0x7ffff7f39e90 <jerry_global_heap+5744>)
    at ../src/loaders/lottie/jerryscript/jerry-core/ecma/base/ecma-gc.cpp:1543
#2  ecma_gc_free_properties (object_p=object_p@entry=0x7ffff7f39e90 <jerry_global_heap+5744>,
    options=options@entry=0) at ../src/loaders/lottie/jerryscript/jerry-core/ecma/base/ecma-gc.cpp:1658
#3  0x00007ffff7e4b024 in ecma_gc_free_object (object_p=0x7ffff7f39e90 <jerry_global_heap+5744>)
    at ../src/loaders/lottie/jerryscript/jerry-core/ecma/base/ecma-gc.cpp:1696
#4  ecma_gc_run () at ../src/loaders/lottie/jerryscript/jerry-core/ecma/base/ecma-gc.cpp:2210
#5  0x00007ffff7e4b67d in ecma_free_unused_memory (pressure=pressure@entry=JMEM_PRESSURE_LOW)
    at ../src/loaders/lottie/jerryscript/jerry-core/ecma/base/ecma-gc.cpp:2245
#6  0x00007ffff7e0a262 in jmem_heap_gc_and_alloc_block (max_pressure=JMEM_PRESSURE_FULL, size=16)
    at ../src/loaders/lottie/jerryscript/jerry-core/jmem/jmem-heap.cpp:292
#7  jmem_heap_alloc_block (size=size@entry=16)
    at ../src/loaders/lottie/jerryscript/jerry-core/jmem/jmem-heap.cpp:335
#8  0x00007ffff7e48f0e in ecma_alloc_property_pair ()
    at ../src/loaders/lottie/jerryscript/jerry-core/ecma/base/ecma-alloc.cpp:215
#9  0x00007ffff7e534d9 in ecma_create_property (out_prop_p=0x7fffffffd7e8, value=..., type_and_flags=32 ' ',
    name_p=0x257d, object_p=0x7ffff7f3a7f8 <jerry_global_heap+8152>)
    at ../src/loaders/lottie/jerryscript/jerry-core/ecma/base/ecma-helpers.cpp:454
#10 ecma_create_named_data_property (object_p=object_p@entry=0x7ffff7f3a7f8 <jerry_global_heap+8152>,
    name_p=name_p@entry=0x257d, prop_attributes=prop_attributes@entry=0 '\000',
    out_prop_p=out_prop_p@entry=0x7fffffffd7e8)
    at ../src/loaders/lottie/jerryscript/jerry-core/ecma/base/ecma-helpers.cpp:533
#11 0x00007ffff7e4d557 in ecma_create_native_pointer_property (obj_p=0x7ffff7f3a7f8 <jerry_global_heap+8152>,
    native_p=0x555555a5b0a0, native_info_p=<optimized out>)
    at ../src/loaders/lottie/jerryscript/jerry-core/ecma/base/ecma-helpers-external-pointers.cpp:64
#12 0x00007ffff7eb3e4c in _buildLayer (context=363, frameNo=0, layer=0x555555645df0, comp=0x555555b25160,
    exp=<optimized out>) at ../src/loaders/lottie/tvgLottieExpressions.cpp:67
#13 0x00007ffff7eb5ac8 in LottieExpressions::evaluate (this=0x555555a529f0, frameNo=0, exp=0x555555aaaff0)
    at ../src/loaders/lottie/tvgLottieExpressions.cpp:1316
#14 0x00007ffff7eb0337 in LottieExpressions::result<LottieGenericProperty<tvg::Point> > (exp=0x555555aaaff0,
    out=<synthetic pointer>..., frameNo=0, this=0x555555a529f0)
    at ../src/loaders/lottie/tvgLottieExpressions.h:57
#15 LottieGenericProperty<tvg::Point>::operator() (this=0x555555645f78, frameNo=0, exps=0x555555a529f0)
    at ../src/loaders/lottie/tvgLottieProperty.h:401
#16 0x00007ffff7ea7899 in _updateTransform (transform=transform@entry=0x555555645f10, frameNo=frameNo@entry=0,
    autoOrient=<optimized out>, matrix=..., opacity=@0x555555645ef8: 32 ' ', exps=0x555555a529f0)
    at ../src/loaders/lottie/tvgLottieBuilder.cpp:135
#17 0x00007ffff7ea83fe in LottieBuilder::updateTransform (this=this@entry=0x5555559e85d0,
    layer=layer@entry=0x555555645df0, frameNo=frameNo@entry=0) at ../src/loaders/lottie/tvgLottieBuilder.cpp:161
#18 0x00007ffff7ead0b1 in LottieBuilder::updateTransform (frameNo=0, layer=0x555555645df0, this=0x5555559e85d0)
    at ../src/loaders/lottie/tvgLottieBuilder.cpp:152
#19 LottieBuilder::updateLayer (this=0x5555559e85d0, comp=0x555555a4da90, scene=0x5555555ded70,
    layer=0x555555645df0, frameNo=0) at ../src/loaders/lottie/tvgLottieBuilder.cpp:1217
#20 0x00007ffff7ead5e9 in LottieBuilder::updatePrecomp (frameNo=<optimized out>, precomp=0x5555555de100,
    comp=0x555555a4da90, this=0x5555559e85d0) at ../src/loaders/lottie/tvgLottieBuilder.cpp:985
--Type <RET> for more, q to quit, c to continue without paging--
#21 LottieBuilder::updatePrecomp (this=0x5555559e85d0, comp=0x555555a4da90, precomp=0x5555555de100,
    frameNo=<optimized out>) at ../src/loaders/lottie/tvgLottieBuilder.cpp:977
#22 0x00007ffff7eadbd4 in LottieBuilder::updateLayer (frameNo=0, layer=0x5555555de100, scene=0x5555555de690,
    comp=0x555555a4da90, this=0x5555559e85d0) at ../src/loaders/lottie/tvgLottieBuilder.cpp:1235
#23 LottieBuilder::updateLayer (frameNo=0, layer=0x5555555de100, scene=0x5555555de690, comp=0x555555a4da90,
    this=0x5555559e85d0) at ../src/loaders/lottie/tvgLottieBuilder.cpp:1210
#24 LottieBuilder::update (this=this@entry=0x5555559e85d0, comp=comp@entry=0x555555a4da90,
    frameNo=frameNo@entry=0) at ../src/loaders/lottie/tvgLottieBuilder.cpp:1389
#25 0x00007ffff7eadcb8 in LottieBuilder::build (this=0x5555559e85d0, comp=0x555555a4da90)
    at ../src/loaders/lottie/tvgLottieBuilder.cpp:1404
#26 0x00007ffff7eb75fd in LottieLoader::run (this=0x555555a4df50, tid=<optimized out>)
    at ../src/loaders/lottie/tvgLottieLoader.cpp:46
#27 0x00007ffff7eb7c29 in LottieLoader::header (this=0x555555a4df50)
    at ../src/loaders/lottie/tvgLottieLoader.cpp:79
#28 0x00007ffff7e0caf3 in LoaderMgr::loader (path="/home/hermet/thorvg/examples/resources/lottie/sample.json",
    invalid=invalid@entry=0x7fffffffdd7f) at ../src/renderer/tvgLoader.cpp:307
#29 0x00007ffff7e10c6d in tvg::Picture::Impl::load (path="", this=0x555555b07000)
    at ../src/renderer/tvgPicture.h:130
#30 tvg::Picture::Impl::load (path="", this=0x555555b07000) at ../src/renderer/tvgPicture.h:125
#31 tvg::Picture::load (this=this@entry=0x5555558dcee0,
    path="/home/hermet/thorvg/examples/resources/lottie/sample.json") at ../src/renderer/tvgPicture.cpp:166
#32 0x000055555555799a in UserExample::content (h=<optimized out>, w=1024, canvas=0x555555b1db90,
    this=<optimized out>) at ../examples/Animation.cpp:46
#33 tvgexam::Window::ready (this=0x55555556df00) at ../examples/Example.h:183
#34 0x00005555555567ee in tvgexam::main (example=<optimized out>, argc=argc@entry=1,
    argv=argv@entry=0x7fffffffe008, width=width@entry=1024, height=height@entry=1024,
    threadsCnt=threadsCnt@entry=4, print=false) at ../examples/Example.h:532
#35 0x0000555555556b37 in main (argc=1, argv=0x7fffffffe008) at ../examples/Animation.cpp:93
```

**Comment by hermet** [comment]
@theashraf Hello, the crash will be fixed in v0.14.8, but the effect still won't work afterward due to the missing layer effect. Please keep this: https://github.com/thorvg/thorvg/issues/2153
